pkg/authz/authz.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Abstract Machines
+// SPDX-License-Identifier: Apache-2.0
+
+package authz
+
+import "context"
+
+type PolicyReq struct {
+	// Domain contains the domain ID.
+	Domain string `json:"domain,omitempty"`
```

Comment:
We have exactly the same structure in `pkg/policies/manager.go`. Overall, it looks like we have too many abstractions and you got tangled in them. Please simplify.

Reply:
Lets take this case 

https://github.com/absmach/magistrala/pull/2453#discussion_r1799589120
> Let's rename to AuthNReq, AuthNResp. Do the same for AuthZReq and AuthZResp.

Now we need to make change only in `pkg/authz` and `pkg/authn` not in all service

Reply:
OK, let's rename Manager to Service (it's policy CRUD service) and `PolicyReq` to `Policy` in `pkg/policies/manager.go`. Also, consider moving that to `internal` rather than `pkg` because Policy CRUD service is meant to be internal (the client, on the other hand, could be public).